### DISCUSSION TITLE
Waveshare ESP32-S3-Touch-LCD-4.3

### DISCUSSION THREAD

**Discussion by Westcott1** [moononournation/Arduino_GFX#409] [Show and tell]
Hello,

I've managed to get one of these working.
Even the LvglWidgets example.

No idea what to set the xsync_xxxx values to!
Also no idea what the backlight pin  and touch pins are.
[https://www.waveshare.com/wiki/ESP32-S3-Touch-LCD-4.3](url)

#define GFX_DEV_DEVICE WAVESHARE_ESP32_S3_TFT_4_3
//define GFX_BL 2
Arduino_ESP32RGBPanel *rgbpanel = new Arduino_ESP32RGBPanel(
    5 /* DE */,
    3 /* VSYNC */,
    46 /* HSYNC */,
    7 /* PCLK */,
    14 /* B0 */,
    38 /* B1 */,
    18 /* B2 */,
    17 /* B3 */,
    10 /* B4 */,
    39 /* G0 */,
    0 /* G1 */,
    45 /* G2 */,
    48 /* G3 */,
    47 /* G4 */,
    21 /* G5 */,
    1 /* R0 */,
    2 /* R1 */,
    42 /* R2 */,
    41 /* R3 */,
    40 /* R4 */,
    0 /* hsync_polarity */, 40 /* hsync_front_porch */, 48 /* hsync_pulse_width */, 88 /* hsync_back_porch */,
    0 /* vsync_polarity */, 13 /* vsync_front_porch */, 3 /* vsync_pulse_width */, 32 /* vsync_back_porch */,
    1 /* pclk_active_neg */, 16000000 /* prefer_speed */);

Arduino_RGB_Display *gfx = new Arduino_RGB_Display(
    800 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */);

**Comment by moononournation** [comment]
Is it display ok for the above setting? If not, you may refer to the display datasheet values.

**Comment by Westcott1** [comment]
Hi Moononournation,
Yes, I soon realised which are the touch pins.
SDA=8, SCL=9, IRQ=4,
However, touch RST is expansion pin 1,  I've not got the expansion pins to work yet.
They also say you can use one of the expansion pins as Backlight control.

I thing I might start a Github repository devoted to this particular board.

Also, thanks for developing Arduino_gfx!

**Reply by jalmince** [reply]
Hi Westcott1 !
I ve got the same board from WaveShare, but I m struggling to get it to work. I wish I could port a project that runs on the too small Lilygo t-display with tft_espi ( no lvgl ) . Would you be so nice as to send me a code example ? ( or link to your github if you ve done it ? )
Cheers,
Jean-Paul

**Reply by Westcott1** [reply]
Hi Jalmince,
I'll create an example using TFT_eSPI for you.
The important bit it to get the pins right!

**Reply by jalmince** [reply]
Thanks!
To be sure we  re talking the same board, mine is https://www.waveshare.com/catalogsearch/result/?q=25948
cheers,
Jpaul

**Reply by Westcott1** [reply]
Yes, that's the one - nice board and display  at a good price.
Took me ages to get everything to run.

**Reply by Westcott1** [reply]
After quite a bit of digging, it seems that TFT_eSPI does NOT support 16bit parallel displays on ESP32.
This must be why the Waveshare demo code uses their own ESP_Panel_Library.

**Comment by jalmince** [comment]
Hi Amish! 
Thnks for your answer.

I know tft_spi doesn't support 16 parallel display, otherwise I wouldn't have problems to port my code from the Lilygo t-Display on the waveshare board.
However you said you managed to get the WaveShare board to work with this Arduino_GFX  lib.

What is it exactly to manage to run ?

If, as said here in the README :

_This library start rewrite from Adafruit_GFX, LovyanGFX, **TFT_eSPI**, Ucglib, and more...
Simple Declaration
#include <Arduino_GFX_Library.h>
Arduino_DataBus *bus = new Arduino_HWSPI(16 /* DC */, 5 /* CS */);
Arduino_GFX *gfx = new Arduino_ILI9341(bus, 17 /* RST */);
**And Simple Usag**e
gfx->begin();
gfx->fillScreen(BLACK);
gfx->setCursor(10, 10);
gfx->setTextColor(RED);
gfx->println("Hello World!");_

Then that ' s all I am after !  
I should not even have to change the graphical primitives names ( fillScreen, DrawCircle,....) 

the only thing is I don t want to use Panel and lvgl.

So do you have code that draw and displays text on the screen ?!

cheers,
Jean-Paul

**Reply by Westcott1** [reply]
Hi Jean-Paul,
Ah, I understand now!
Yes, I added an Arduino_GFX "Hello World" example yesterday.
It includes touch and expander pins as well, if you want them.
Regards,
Hamish
(C'est un prénom écossais)

**Comment by jalmince** [comment]
Hi Amish!

Did you add it to a place I have access to ?

Ps: my daughter and ex-wife live in Edinburgh.
I  lived there for 3 years but now I am in Switzerland


Le dim. 18 févr. 2024, 12:43, Hamish West ***@***.***> a
écrit :

**Reply by Westcott1** [reply]
[Arduino_gfx example](https://github.com/Westcott1/Waveshare-ESP32-S3-Touch-LCD-4.3-and-Arduino/tree/main/Examples/Arduino_GFX)
Lovely city Edinburgh.

**Comment by jalmince** [comment]
Thanks mate, you made my day!

Le dim. 18 févr. 2024, 17:39, Hamish West ***@***.***> a
écrit :
